src/validation.cairo

```diff
@@ -56,8 +60,41 @@ fn next_prev_timestamps(self: @ChainState, block: @Block) -> Span<u32> {
 }
 
 fn compute_total_work(self: @ChainState, block: @Block) -> u256 {
-    // TODO: implement
-    *self.total_work
+    let block_target: u256 = convert_u32_to_u256(*block.header.bits);
```

Comment:
you need either to use bits_to_target function which is not jet finished or assume that target is given as parameter to compute_total_work

Reply:
Sure, I'll update the PR to use the bits_to_target function when it's available.

Reply:
It makes more sense to change function signature to accept target as parameter instead of `ChainState`. This will allow us to merge the pr sooner.

Reply:
should the function also accept the chainstate.total_work member as a parameter? ChainState type is accepted as a parameter in order to access the value of total_work

Reply:
https://github.com/keep-starknet-strange/raito/blob/f599f8124e7acf0cf4644a262fada94f54d20513/src/validation.cairo#L17 uses the compute_total_work(self: @ChainState, block: @Block) function signature. Am I good to modify it?

Reply:
yes, please

Reply:
modified the function signature